lib/koans/10_processes.ex

```diff
@@ -26,15 +45,71 @@ defmodule Processes do
   end
 
   koan "A common pattern is to include the sender in the message, so that it can reply" do
-    pid = spawn(fn -> receive do
-                         {:hello, sender} -> send sender, :how_are_you?
-                      end
-                 end)
+    greeter = fn ->
+      receive do
+        {:hello, sender} -> send sender, :how_are_you?
+      end
+    end
+
+    pid = spawn(greeter)
 
     send pid, {:hello, self()}
     assert_receive ___
   end
 
+  def yelling_echo_loop do
+    receive do
+      {caller, value} ->
+        send caller, String.upcase(value)
+        yelling_echo_loop
+    end
+  end
+
+  koan "Use tail recursion (calling a function as the very last statement) to receive multiple messages" do
```

Comment:
I can't really think of a way to teach about tail recursion using koans. The only visible effect of it is faster runtime and bounded memory usage – not really something we can make nice assertions about.

I see the point of at least mentioning tail recursion, though. It is an important concept and something that every functional programmer should understand. Maybe just throwing the term out there is enough. If users want to dig deeper they can go off on their own.

Could we at least get rid of the text in parentheses? To me the real lesson of the koan is that by recursing you can keep responding to messages. I'm afraid that the text in parentheses may distract users to think that the koan is _about_ tail recursion itself rather than the `receive -> recurse` idiom.